wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java

```diff
@@ -149,20 +166,36 @@ public Connection getConnection(final String username, final String password) th
       }
     }
 
+    TargetDriverDialect targetDriverDialect = configurationProfile == null
+        ? null
+        : configurationProfile.getTargetDriverDialect();
+
     // Identify what connection provider to use.
     if (!StringUtils.isNullOrEmpty(this.targetDataSourceClassName)) {
 
       final DataSource targetDataSource = createTargetDataSource();
-      final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
-      final TargetDriverDialect targetDriverDialect =
-          targetDriverDialectManager.getDialect(this.targetDataSourceClassName, props);
+
+      if (targetDriverDialect == null) {
+        final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
+        targetDriverDialect = targetDriverDialectManager.getDialect(this.targetDataSourceClassName, props);
+      }
+
+      ConnectionProvider defaultConnectionProvider = new DataSourceConnectionProvider(
+          targetDataSource,
+          targetDriverDialect);
+
+      ConnectionProvider effectiveConnectionProvider =
```

Comment:
Can `effectiveConnectionProvider` be initialized where targetDriverDialect was initialized to reduce code duplication

Reply:
Not sure I understand what piece of code do you mean.